setup.cfg

```diff
@@ -3,7 +3,7 @@ requirement_cls = cockroachdb.sqlalchemy.test_requirements:Requirements
 profile_file=.profiles.txt
 
 [db]
-default=cockroachdb://root@localhost:26257/defaultdb
+default=cockroachdb://root@localhost:26257/defaultdb?disable_cockroachdb_telemetry=True
```

Comment:
just curious -- how does this work? what reads this setting?

Reply:
https://docs.sqlalchemy.org/en/14/core/engines.html#add-parameters-to-the-url-query-string

From what I understand, the connect_args params gets created from the URL#143 
I can't cc gordthompson in this repo but I think he probably has a better understanding.

Reply:
ah interesting; i thought the params here get passed to the database -- i thought they would be used to set session variables like application_name

Reply:
If you're asking how SQLA parses the connection URI string and converts it into a `URL` object the fun starts here:

https://github.com/sqlalchemy/sqlalchemy/blob/585aa7d7575dfb9a90a00bbd00c68da9e2cb31d7/lib/sqlalchemy/engine/url.py#L686

Does that help?

Reply:
Example: A connection URI like

```
mssql+pyodbc://localhost:49242/mydb?driver=ODBC+Driver+17+for+SQL+Server
```

gets converted to a `URL` object containing (among a few other things)

```
0 = {str} 'mssql+pyodbc'
1 = {NoneType} None
2 = {NoneType} None
3 = {str} 'localhost'
4 = {int} 49242
5 = {str} 'mydb'
6 = {immutabledict: 1} immutabledict({'driver': 'ODBC Driver 17 for SQL Server'})
```

The dialect can then pick entries out of that last item and do useful stuff with them.